applications/examples/GradientDamageCalibration.cpp

```diff
@@ -0,0 +1,176 @@
+//! @remark This method may fail due to various reasons:
+//!   - boundaryDisplacement not big enough (structure does not fully unload)
+//!   - combination of c, gf (local fracture energy parameter) and L causes a snap-back
+//!   - ???
+template <typename TGdm>
+double GlobalFractureEnergy(TGdm& gdm, double L = 50, int nElements = 200, double boundaryDisplacement = 0.1)
+{
+    DofType d = gdm.mDisp;
+    ScalarDofType eeq = gdm.mEeq;
+
+    MeshFem mesh = UnitMeshFem::Transform(UnitMeshFem::CreateLines(nElements),
+                                          [&](Eigen::VectorXd x) { return Eigen::VectorXd::Constant(1, x[0] * L); });
+
+    InterpolationTrussLobatto interpolationD(2);
+    AddDofInterpolation(&mesh, d, interpolationD);
+    AddDofInterpolation(&mesh, eeq);
+
+    Constraint::Constraints constraints;
+    constraints.Add(d, Constraint::Component(mesh.NodesAtAxis(eDirection::X, d), {eDirection::X}));
+    constraints.Add(d, Constraint::Component(mesh.NodesAtAxis(eDirection::X, d, L), {eDirection::X},
+                                             Constraint::RhsRamp(1, boundaryDisplacement)));
+
+    IntegrationTypeTensorProduct<1> integration(3, eIntegrationMethod::GAUSS);
+    const int nIp = integration.GetNumIntegrationPoints();
+    CellStorage cellStorage;
+    auto cells = cellStorage.AddCells(mesh.ElementsTotal(), integration);
+
+    gdm.mKappas.setZero(cells.Size(), nIp);
+    gdm.mKappas.row(cells.Size() / 2) = Eigen::VectorXd::Constant(nIp, 3 * k0);
+    gdm.mKappas.row(cells.Size() / 2 + 1) = Eigen::VectorXd::Constant(nIp, 3 * k0);
+
+    TimeDependentProblem equations(&mesh);
+    equations.AddGradientFunction(cells, TimeDependentProblem::Bind(gdm, &TGdm::Gradient));
+    equations.AddHessian0Function(cells, TimeDependentProblem::Bind(gdm, &TGdm::Hessian0));
+    equations.AddUpdateFunction(cells, TimeDependentProblem::Bind(gdm, &TGdm::Update));
+
+    QuasistaticSolver problem(equations, {d, eeq});
+    problem.SetQuiet();
+    problem.mTolerance = 1.e-6;
+    problem.SetConstraints(constraints);
+
+    int dofLeft = mesh.NodeAtCoordinate(EigenCompanion::ToEigen(L), d).GetDofNumber(0);
+
+    auto loadDispFileName = "CalibrationLD.dat";
+    std::ofstream loadDisplacement(loadDispFileName);
+    auto doStep = [&](double t) { return problem.DoStep(t, "MumpsLU"); };
+    auto postProcessF = [&](double) { problem.WriteTimeDofResidual(loadDisplacement, d, {dofLeft}); };
+
+    AdaptiveSolve adaptiveSolve(doStep, postProcessF);
+    adaptiveSolve.SetQuiet();
+    adaptiveSolve.dt = 0.01;
+    adaptiveSolve.dtMin = 1.e-10;
+    adaptiveSolve.dtMax = 0.01;
+    adaptiveSolve.Solve(1.);
+
+    loadDisplacement.close();
+    Eigen::VectorXd loads = EigenIO::ReadFromFile(loadDispFileName).col(2);
+    Eigen::VectorXd disps = EigenIO::ReadFromFile(loadDispFileName).col(1);
+
+    Tools::GlobalFractureEnergyIntegrator gfIntegrator(loads, disps);
+    double GfTotal = gfIntegrator.IntegrateSofteningCurve(1., 0.1);
```

Comment:
1. What is this »"crack" area«? The cross-section of the specimen?
2. I assume `rForceThreshold` is in "force units". Would it make sense to supply that in "percent of maximum force"?